src/QuadForm/Herm/Lattices.jl

```diff
@@ -750,3 +750,114 @@ function maximal_integral_lattice(V::HermSpace)
   return maximal_integral_lattice(L)
 end
 
+################################################################################
+#
+#  Intersection
+#
+################################################################################
+
+#=
+    _reconstruction_herm_lattice(L::ZLat, E::Hecke.NfRel{nf_elem}, 
+    f::Hecke.VecSpaceRes{Hecke.NfRel{nf_elem}, Hecke.NfRelElem{nf_elem}}, Vas)
+                  -> HermLat{} 
+    
+Takes the $\mathbb{Z}$-lattice $L$ and reconstructs a herm lattice with base field
+$E$ and ambient space $Vas$. 
+=#
+function _reconstruction_herm_lattice(L::ZLat, E::Hecke.NfRel{nf_elem}, f::Hecke.VecSpaceRes{Hecke.NfRel{nf_elem}, Hecke.NfRelElem{nf_elem}}, Vas)
```

Comment:
I guess this should only take as entry a matrix, the basis matrix of `L` namely and the function `f`... Indeed in intersection, you construct a lattice from a basis matrix and here you ask directly for the basis matrix: you could only keep track of the basis matrix :-) As for `E` and `Vas`, they are obtained via `f`: one should be able to obtain them from `domain` and `codomain`... You should add functions in the corresponding files that does it, we can discuss it later.